sys-boot/tboot/tboot-1.10.3.ebuild

```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit mount-boot toolchain-funcs
+
+DESCRIPTION="Performs a measured and verified boot using Intel Trusted Execution Technology"
+HOMEPAGE="https://sourceforge.net/projects/tboot/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE="custom-cflags selinux"
+
+# requires patching the kernel src
+RESTRICT="test"
+
+DEPEND="dev-libs/openssl:0=[-bindist(-)]"
+
+RDEPEND="${DEPEND}
+	sys-boot/grub:2
+	selinux? ( sec-policy/selinux-tboot )"
+
+DOCS=( README.md COPYING CHANGELOG )
+PATCHES=( "${FILESDIR}/${PN}-1.10.3-disable-Werror.patch"
+	  "${FILESDIR}/${PN}-1.10.3-disable-strip.patch" )
+
+src_prepare() {
+	sed -i \
+		-e "s/\(AS[[:space:]]\{1,\}\)?= as/\1= $(tc-getAS)/" \
+		-e "s/\(LD[[:space:]]\{1,\}\)?= ld/\1= $(tc-getLD)/" \
+		-e "s/\(CC[[:space:]]\{1,\}\)?= gcc/\1= $(tc-getCC)/" \
+		-e "s/\(CPP[[:space:]]\{1,\}\)?= cpp/\1= $(tc-getCPP)/" \
+		-e "s/\(AR[[:space:]]\{1,\}\)?= ar/\1= $(tc-getAR)/" \
+		-e "s/\(RANLIB[[:space:]]\{1,\}\)?= ranlib/\1= $(tc-getRANLIB)/" \
+		-e "s/\(NM[[:space:]]\{1,\}\)?= nm/\1= $(tc-getNM)/" \
+		-e "s/\(STRIP[[:space:]]\{1,\}\)?= strip/\1= $(tc-getSTRIP)/" \
+		-e "s/\(OBJCOPY[[:space:]]\{1,\}\)?= objcopy/\1= $(tc-getOBJCOPY)/" \
+		-e "s/\(OBJDUMP[[:space:]]\{1,\}\)?= objdump/\1= $(tc-getOBJDUMP)/" \
```

Comment:
these are all `?=` defined, you can override them from outside without any patching:
`tc-export AS LD CC CPP AR RANLIB NM OBJCOPY OBJDUMP`
`LD` is a bit cheesy, seeing you generally don't need to invoke this during a build